python/cudf/cudf/tests/test_array_ufunc.py

```diff
@@ -23,11 +34,77 @@ def test_ufunc_cudf_series(np_ar_tup, func):
         assert_eq(expect, got.to_array())
 
 
+@pytest.mark.parametrize(
+    "func", [np.bitwise_and, np.bitwise_or, np.bitwise_xor],
+)
+def test_ufunc_cudf_series_bitwise(func):
+    x = np.random.randint(size=100, low=0, high=100)
+    y = np.random.randint(size=100, low=0, high=100)
+
+    s_1, s_2 = cudf.Series(x), cudf.Series(y)
+    expect = func(x, y)
+    got = func(s_1, s_2)
+    if np.isscalar(expect):
+        assert_eq(expect, got)
+    else:
+        assert_eq(expect, got.to_array())
+
+
 @pytest.mark.parametrize(
     "np_ar_tup", [(np.random.random(100), np.random.random(100))]
 )
 @pytest.mark.parametrize(
-    "func", [np.greater, np.less, np.less_equal],
+    "func",
+    [
+        np.subtract,
+        np.multiply,
+        np.floor_divide,
+        np.true_divide,
+        np.power,
+        np.remainder,
+        np.divide,
+        np.equal,
+        np.not_equal,
+        np.less,
+        np.less_equal,
+        np.greater,
+        np.greater_equal,
+    ],
+)
+def test_ufunc_cudf_series_cudf_dispatch(np_ar_tup, func):
+    x, y = np_ar_tup[0].astype(np.float32), np_ar_tup[1].astype(np.float32)
+    x[0] = np.nan
+    y[1] = np.nan
+    s_1, s_2 = cudf.Series(x), cudf.Series(y)
+    expect = func(x, y)
+    got = func(s_1, s_2)
+    got = got.fillna(np.nan)
```

Comment:
I think that this succeeds for a few of the comparison functions due to a bug in cuDF. 

```
x = cudf.Series(np.array([1, 2, np.nan])
y = cudf.Series(np.array([1, 2, np.nan])
```

The `nan` gets converted to a true `<NA>`, giving me

```
>>> y
0     1.0
1     2.0
2    <NA>
dtype: float64
```

Then doing a comparison between x and y would get me, for `>` for instance, 

```
>>> x > y
0    False
1    False
2    False
dtype: bool
```
The third entry should be an `<NA>` here. Currently when `fillna` is called then, it's a no-op for the cuDF series. But the numpy side of things gives us `array([False, False, False])` because it's comparing `NaN` rather than `<NA>`, so the comparison passes.

I think what we're supposed to be getting is 

```
>>> x > y
0    False
1    False
2    <NA>
dtype: bool
```

And then we'd ultimately be comparing `array([False, False, False])` to a cuDF series containing `[False, False, NaN]`. 

This should either be xfailed and be allowed to XPASS or have some conditional handling. 

@kkraus14 @shwina

Reply:
I'm not sure I'm following. What's the numpy result (`expect`) here?

If we want to test nan comparison behavior rather than null comparison behaviour, we should construct the cudf test input with `nan_as_null=False`. If we want to test null comparison behaviour, I don't think we can test against NumPy - we probably have to write our own expectation. Does that sound about right @brandon-b-miller?

Reply:
basically we just have the wrong result in cuDF currently for comparison ops involving `<NA>`. 

```
>>> gsr = cudf.Series([1,2,None])
>>> gsr > gsr
0    False
1    False
2    False
dtype: bool
```

Reply:
@brandon-b-miller ,

For numpy we get below behavior which I think is what we want for comparison ops involving <NA>, are we planning on deviating from that behavior cause that will be concerning.   
```python
>>> import numpy as np
>>> ar = np.array([1,2,np.nan])
>>> ar > ar
array([False, False, False])
```

Reply:
We want `nan` to follow the numpy comparison semantics. I think that works fine. 
```
>>> gsr = cudf.Series([1,2,None], dtype='float32')
>>> gsr
0     1.0
1     2.0
2    <NA>
dtype: float32
>>> gsr[2] = np.nan
>>> gsr
0    1.0
1    2.0
2    NaN
dtype: float32
>>> gsr > gsr
0    False
1    False
2    False
dtype: bool
```

But we want `<NA>` > `<NA>` to equal `<NA>` and currently it's returning `False`. I believe if it worked as such than the last element of the results wouldn't compare correctly.

Reply:
I think I get the confusion here, In this test I wanted to check that if we have `None` in our series it gets operated upon successfully for these binary ops and while writing the test I relied on the fact that we by default switch our nans to nulls, which i should be more explicit about.  

That said I still don't get why `<NA>` > `<NA>` should be equal to `<NA>`.  This is where my confusion is stemming from.  I think it should be equal to False.  

  ```python
>>> gsr = cudf.Series([1,None,2], dtype='float32')
>>> gsr > gsr
0    False
1    False
2    False
dtype: bool
````

Reply:
Had a chat with @brandon-b-miller , will add the comparison tests will nulls to a different test and mark them as xfail.

Reply:
Marked comparison operations as xfail, resolved by https://github.com/rapidsai/cudf/pull/6973/commits/4be146433912fed09e094c81ea4881e7b7921df6